### DISCUSSION TITLE
Slow Downloads (Solved)

### DISCUSSION THREAD

**Discussion by musicaltank** [vicwomg/pikaraoke#259] [Q&A] [answered]
My downloads seem to be very slow, and may be a network issue. I'm running Pikaraoke version 1.1.2 and yt-dlp version 2022.04.08 on a Rasberry Pi 4B. Is there a menu option where you can view a list of current active downloads and/or progress? OR Is there a way to verify network connectivity that I'm missing?

**Comment by vicwomg** [comment]
Update yt-dl and see if it improves. Your version is over a year old

**Reply by musicaltank** [reply]
I can't seem to get yt-dlp to update using the link in the "Pikaraoke" submenu. I'm not Pi/Linux illiterate, but I'm no expert either - is there another way to update? Can I access the terminal in Pikaraoke? The network seems to be OK - I can obviously access the web interface, and none of my other systems seem to have issues accessing internet.

**Comment by vicwomg** [comment]
Pull the latest version of pikaraoke. There was a big in the updater

**Comment by vicwomg** [comment]
To be clear, to update pikaraoke, you need to log into your pi. CD to the
pikaraoke directory, and run "git pull" then restart

**Reply by musicaltank** [reply]
Ahh ok, thank you for clarifying - I was confused because I had used your latest image (pikaraoke-1.1.2.img) from May 10, 2022. I'll try to log into the Pi - my problem is my fingers aren't fast enough to log in on startup before it moves on to the splash screen :D

**Reply by musicaltank** [reply]
OK, I sucessfully logged into the pi, CD'd to the pikaraoke directory, ran "git pull", and then restarted. I still seem to have yt-dlp version 2022.04.08.

![Pikaraoke system info](https://github.com/vicwomg/pikaraoke/assets/118386938/a840aa3f-1b6c-432e-ad69-f86f3d9a6fac)

**Comment by vicwomg** [comment]
After updating, then try running the yt-dl updater , to be clear that fixes
the issue where it didn't work.

**Reply by musicaltank** [reply]
Unfortunately, I ran the yt-dlp updater and there was no change - still version 2022.04.08

**Comment by vicwomg** [comment]
My sense is you didn't upgrade pikaraoke. What was the output to the git
pull?

In any case the issue is that yt-dl is outdated. If you can get to the
console you can also upgrade it there. The command to do so is in the
readme or easily looked up

**Reply by musicaltank** [reply]
Thanks, very much appreciated. I should have done a screen grab of the output, but I've probably closed that terminal by now. I don't remember exactly, but it looked like it updated about 9 different things. I ran it a second time just for fun, and that time said the system was already up to date - details aside, the outputs were very different from the first to second pull. I'll go through the Readme again and/or do more searching on updating yt-dlp. When I get it working, I'll post an update here.

This looks like an awesome project, and I really appreciate your time to develop, and answer my questions here. I'm hoping to get it working by this weekend for my daughters' joint birthday party - they wanted the theme to be the Trolls movies, so karaoke would be a perfect activity. Thanks again.

**Reply by musicaltank** [reply]
So, I've updated yt-dlp using:
pip3 install --upgrade yt-dlp

And verified the version using:
yt-dlp -U

Which outputs:
![20230517_213734](https://github.com/vicwomg/pikaraoke/assets/118386938/e216d000-e3b6-401c-beba-dca1f4b2e455)

But when I go into the web interface, it still lists the 2022 version.

At this point, I might just reflash the SD with a fresh image, in case I've somehow messed something up along the way, and then try to update again.

**Comment by vicwomg** [comment]
What's happening here is you have two versions installed. Pikaraoke is
using the other one.

Try upgrading with python package manager
sudo pip3 install yt-dl -U

**Comment by vicwomg** [comment]
Sorry just read you used pip3 after all. After you upgraded it, did you
restart pikaraoke?

**Comment by vicwomg** [comment]
On third look, the syntax to your pip3 command looks incorrect. Try mine.
And pay attention to the output to confirm success

**Reply by musicaltank** [reply]
Awesome, thanks. I'll give it a try tonight and capture the output.

**Reply by musicaltank** [reply, marked as answer]
Success! You were correct, I needed to use your syntax to update with the python package manager, with one small edit - I had to use "yt-dlp" instead of "yt-dl". For clarity, here is the command that fixed the issue:

sudo pip3 install yt-dlp -U

And the output in terminal:
![20230519_091137](https://github.com/vicwomg/pikaraoke/assets/118386938/5469db3e-8d24-498a-b945-3f5345f2487e)

And verification that the correct installation/version was updated and recognized by pikaraoke from the web interface:
![Screenshot_20230519_092556_Chrome](https://github.com/vicwomg/pikaraoke/assets/118386938/e7d11b76-32ed-493e-a295-0a430b0b6b86)

Now all I need is for my microphones to be delivered and we should be up and running.

Thanks again for this project, and your support troubleshooting!
